bot/telegram/telegram_bot.go

```diff
@@ -43,6 +48,40 @@ func (b *Bot) Serve() error {
 
 	updates := bot.GetUpdatesChan(u)
 
+	location, err := time.LoadLocation(constants.TimeZone)
+	if err != nil {
+		log.Fatalf("Failed to load location: %v", err)
+	}
+
+	c := cron.New(cron.WithLocation(location))
+
+	_, err = c.AddFunc(b.cfg.CronJobRandomImage, func() {
+		for _, idChat := range b.cfg.WhileLists {
+			idNeedToSend, err := strconv.ParseInt(idChat, 10, 64)
+			if err != nil {
+				logger.Errorf("Invalid chat ID: %v", err)
+				continue
+			}
+
+			err = b.SendImageRandomDaily(idNeedToSend)
+			if err != nil {
+				logger.Errorf("Failed to send image to chat ID %d: %v", idNeedToSend, err)
+			} else {
+				logger.Infof("Daily image sent successfully to chat ID %d.", idNeedToSend)
+			}
+		}
+	})
+
+	if err != nil {
+		logger.Errorf("Error scheduling cron job: %v", err)
```

Comment:
should it return err here to finish the function?

Reply:
If you don't want the cron job error to affect the main bot execution flow, then there's no need to return after logging the error.